Makefile

```diff
@@ -140,7 +140,15 @@ init_upstream:
 			(cd upstream && git submodule update --init && git remote add source git@github.com:hashicorp/terraform-provider-aws.git) ; \
 		fi; \
 
-patch_upstream: init_upstream
+export_upstream_patches: init_upstream
+ifeq ($(shell cd upstream && git rev-parse --is-shallow-repository), false)
+	find upstream-patches -type f -delete
+	cd upstream && \
+		LAST_TAG=$$(git describe --abbrev=0 --tags) && \
```

Comment:
Depends on one's depth of Git expertise, these two lines might warrant a comment. I had to test or look up `describe --abbrev=0` and `...` (as opposed to `..`).